lambdas/nino-check/src/helpers/audit.ts

```diff
@@ -0,0 +1,73 @@
+import { EventBridgeClient, PutEventsCommand } from "@aws-sdk/client-eventbridge";
+import { NinoSessionItem } from "../types/nino-session-item";
+import { AuditUser } from "./nino";
+import { PersonIdentityItem } from "../../../common/src/database/types/person-identity";
+import { AuditConfig } from "../types/input";
+import { marshall } from "@aws-sdk/util-dynamodb";
+
+const eventsClient = new EventBridgeClient();
+
+const auditPrefix = "IPV_HMRC_RECORD_CHECK_CRI";
+
+function buildAuditUser(session: NinoSessionItem): AuditUser {
+  return {
+    govuk_signin_journey_id: session.clientSessionId,
+    ip_address: session.clientIpAddress,
+    session_id: session.sessionId,
+    user_id: session.subject,
+    persistent_session_id: session.persistentSessionId,
+  };
+}
+
+export async function sendRequestSentEvent(
+  auditConfig: AuditConfig,
+  session: NinoSessionItem,
+  personIdentity: PersonIdentityItem,
+  nino: string
+) {
+  const user = buildAuditUser(session);
+
+  await eventsClient.send(
+    new PutEventsCommand({
+      Entries: [
+        {
+          DetailType: "REQUEST_SENT",
+          EventBusName: auditConfig.eventBus,
+          Source: auditConfig.source,
+          Detail: JSON.stringify({
+            auditPrefix,
+            user,
+            deviceInformation: auditConfig.deviceInformation,
+            issuer: auditConfig.issuer,
+            nino,
+            // marshall the person identity for consistency with the step function that came before this Lambda
+            userInfoEvent: { Items: [marshall(personIdentity)], Count: 1 },
```

Comment:
Sorry what is this? I can't see it in the [event catalogue](https://event-catalogue.internal.account.gov.uk/events/IPV_HMRC_RECORD_CHECK_CRI_REQUEST_SENT/), is it for event bridge or something?

Reply:
I copied the structure of this audit event from the step function:

https://github.com/govuk-one-login/ipv-cri-check-hmrc-api/blob/bb8393f4ddc65e1e557affe68c9f63d45c641a10/step-functions/nino_check.asl.json#L202

It passes the result of the person identity fetch into the userInfoEvent key on the REQUEST_SENT event (see it writing to `userInfo` here:

https://github.com/govuk-one-login/ipv-cri-check-hmrc-api/blob/bb8393f4ddc65e1e557affe68c9f63d45c641a10/step-functions/nino_check.asl.json#L129

).

You're right that it doesn't appear to be on the event catalogue you linked though.

Reply:
I'm really confused about what this is, [these are the events we're sending ](https://gds.splunkcloud.com/en-GB/app/gds-050-digital-Identity/search?q=index%3D%22gds_di_development%22%20%22IPV_HMRC_RECORD_CHECK_CRI_REQUEST_SENT%22%20%22user_id%22%20source%3D%22http%3Agds_di_staging_txma-ticf%22&earliest=%40d&latest=now) there's nothing about `userInfoEvent` or `userInfo` in there

Reply:
What does this event emitted by your function look like? Guess we can't tell because event bridge isn't hooked up in dev. Maybe one to check post merge?